test/rpc-integration/runDirectoryTest.sh

```diff
@@ -83,8 +83,16 @@ if [ -d $dir ]; then
             params=""
         fi
         # call rpc-client
-        echo "$client $mode $test $params --expect $dir/response-${testname}.json $*"
-        $client $mode $test $params --expect $dir/response-${testname}.json $*
+        if [ $(basename $server) == "booster-dev" ]; then
+            echo "$client $mode $test $params --expect $dir/response-${testname}.booster-dev $*"
+            $client $mode $test $params --expect $dir/response-${testname}.booster-dev $*
+        elif [ $(basename $server) == "kore-rpc-dev" ]; then
+            echo "$client $mode $test $params --expect $dir/response-${testname}.kore-rpc-dev $*"
+            $client $mode $test $params --expect $dir/response-${testname}.kore-rpc-dev $*
```

Comment:
should we have this branch? Given the contract of the booster-proxy is that it returns the same response as kore-rpc, the only two responses should be for booster-dev or booster-prox/kore-rpc

Reply:
Yes, you're right. However, the responses of  `kore-rpc-booster` and `kore-rpc-dev` do differ:
![Screenshot_2023-09-04_12-05-42](https://github.com/runtimeverification/hs-backend-booster/assets/8296326/67d9fea7-eea2-4436-b183-4a6fe3615d7d).

Do you think if would be easy to harmonize the responses of `kore-rpc-booster` and `kore-rpc-dev`? As I understand, `substitution` is more specialized than `predicate`, i.e. we could alter `kore-rpc`/`kore-rpc-dev` to always return substitution as predicate, just like `kore-rpc-booster` does? Or is it more complicated?

Reply:
given we have kore-rpc-dev which wraps kore-rpc into a version more compatible with the booster, and we only use this for testing, we should just modify the response there

Reply:
I guess having this case is a temporary necessity until we actually get response parity.
We could try to identify "substituting" predicates in booster and return them in the `substitution` field. Then again, what are they really for? Why do we not just substitute term and predicate? (and it seems we do, so the substitution is kind of "informational" for the caller to see what happened to the `X` - see other comment)

Reply:
To clarify: in general, we should `#And` the substitution to the `predicate` field and set the `substitution` field to `Nothing`?